media_driver/linux/common/codec/ddi/media_libva_decoder.cpp

```diff
@@ -312,7 +312,7 @@ VAStatus DdiDecode_StatusReport(PDDI_MEDIA_CONTEXT mediaCtx, CodechalDecode *dec
                     {
                         if (mediaSurfaceHeapElmt != nullptr &&
                                 mediaSurfaceHeapElmt->pSurface != nullptr &&
-                                bo == mediaSurfaceHeapElmt->pSurface->bo)
+                                mediaSurfaceHeapElmt->pSurface == surface)
```

Comment:
it means there are some invalid bo , but surface is still there?

Reply:
yeah, it is. it is crashed when trying to access the invalid bo.

Reply:
May I know which case this will happen? Just be curious.

Reply:
I didn't see that before, but decode in #1227 can reproduce this.

Reply:
I am ok with the change , but I am curious for the bo, why it was destroyed? who destroy it?